.github/workflows/pr-realm-js.yml

```diff
@@ -0,0 +1,391 @@
+name: pr-realm-js
+on: pull_request
+
+jobs:
+  build:
+    name: Build for ${{ matrix.variant.os }} ${{ matrix.variant.arch }}
+    runs-on: ${{ matrix.variant.runner }}
+    env:
+      npm_config_realm_local_prebuilds: ${{github.workspace}}/prebuilds
+      REALM_DISABLE_ANALYTICS: 1
+      ANDROID_NDK_HOME: /usr/local/lib/android/sdk/ndk/21.4.7075529
+    strategy:
+      fail-fast: false
+      matrix:
+        variant:
+          - { os: linux, runner: ubuntu-20.04, arch: x64, artifact-path: prebuilds, test-node: true, test-electron: true }
+          - { os: windows, runner: windows-2019, arch: x64, artifact-path: prebuilds, test-node: true, test-electron: true }
+          - { os: windows, runner: windows-2019, arch: ia32, artifact-path: prebuilds }
+          - { os: android, runner: ubuntu-20.04, arch: x86_64, artifact-path: react-native/android/src/main/jniLibs }
+          - { os: android, runner: ubuntu-20.04, arch: armeabi-v7a, artifact-path: react-native/android/src/main/jniLibs }
+          - { os: android, runner: ubuntu-20.04, arch: arm64-v8a, artifact-path: react-native/android/src/main/jniLibs }
+          - { os: android, runner: ubuntu-20.04, arch: x86, artifact-path: react-native/android/src/main/jniLibs }
+          - { os: darwin, runner: macos-11, arch: x64, artifact-path: prebuilds, test-node: true, test-electron: true }
+          - { os: ios, runner: macos-11, arch: apple, artifact-path: react-native/ios/realm-js-ios.xcframework }
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          submodules: "recursive"
+
+      - name: Setup node version
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+          cache: 'npm'
+          cache-dependency-path: '**/package-lock.json'
+
+      - name: Setup NPM version
+        run: npm install -g npm@7
+
+      - name: Get NPM cache directory
+        id: npm-cache-dir
+        run: echo "::set-output name=dir::$(npm config get cache)"
+
+      - name: Restore NPM cache
+        id: npm-cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.npm-cache-dir.outputs.dir }}
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Cache prebuild artifacts
+        id: cache-prebuild
+        uses: actions/cache@v2
+        env:
+          cache-name: prebuild
```

Comment:
This is pretty roundabout  - why do we need to set an env variable for the step?

Reply:
I'll look into this, but it can probably be removed